### PR TITLE
MULE-14606: Email connector doesn't work with IMAP gmail OOTB

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: src/main/java/org/mule/extension/email/api/attributes/BaseEmailAttributes.java [hunk 1]
```diff
@@ -130,7 +130,7 @@ public BaseEmailAttributes(Message msg) {
   /**
    * @return the unique id of the email in a folder.
    */
-  public abstract long getId();
+  public abstract T getId();
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/email/api/attributes/BaseEmailAttributes.java]: api

File: src/main/java/org/mule/extension/email/internal/AbstractEmailConnection.java [hunk 2]
```diff
@@ -151,18 +151,10 @@ private Properties buildSecureProperties(TlsContextFactory tlsContextFactory) th
       properties.setProperty(protocol.getSslCiphersuitesProperty(), join(cipherSuites, WHITESPACE_SEPARATOR));
     }
 
+    properties.setProperty(protocol.getSslEnableProperty(), "true");
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/email/internal/AbstractEmailConnection.java]: why?

Comment (juandesi) [reply src/main/java/org/mule/extension/email/internal/AbstractEmailConnection.java]: this is required by default if using tls

File: src/main/java/org/mule/extension/email/api/attributes/BaseEmailAttributes.java [hunk 3]
```diff
@@ -43,7 +43,7 @@
  *
  * @since 1.0
  */
-public abstract class BaseEmailAttributes implements Serializable {
+public abstract class BaseEmailAttributes<T> implements Serializable {
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/email/api/attributes/BaseEmailAttributes.java]: @noExtend

**Pull request merged**